src/dataframerow/utils.jl

```diff
@@ -35,19 +35,35 @@ function hashrows_col!(h::Vector{UInt},
 end
 
 # should give the same hash as AbstractVector{T}
-function hashrows_col!(h::Vector{UInt},
-                       n::Vector{Bool},
-                       v::AbstractCategoricalVector,
-                       firstcol::Bool)
-    levs = levels(v)
+function hashrows_col_pool!(h::Vector{UInt},
+                            n::Vector{Bool},
+                            v::AbstractVector,
+                            rp::AbstractVector, # this condition is currently met in all implementations, but is not part of the API
```

Comment:
Do we really need that requirement? AFAICT we only need `length` and iteration, which are now required by the API.

Reply:
we need `firstindex` and that indexing is continuous starting with `firstindex` and ending with `lastindex`.

Reply:
Right, the continuity requirement isn't documented in the docstring right now. That's a reasonable assumption to make, but we could add it for clarity. (It doesn't make a lot of sense to have non contiguous indices when you are required to implement `length`, `eachindex`, `firstindex` and `lastindex`. Basically we want array- or tuple-like objects.)

Reply:
Agreed. I have forgotten that we have added these requirements.